### DISCUSSION TITLE
The Problem with Regeneration

### DISCUSSION THREAD

**Discussion by Zitchas** [endless-sky/endless-sky#7041] [Balance, Mechanics, and Interactions]
Endless Sky has a heavy warship problem. Or rather, it has an effectiveness problem in that as one progresses through the game many (perhaps even most, although currently that designation is mostly anecdotal) feel significant pressure to increase their firepower. While some upgrading is possible, increasing one's firepower is virtually always in the form of:
a) upgrade to a bigger ship
b) add more warships to one's fleet.

This happens for a variety of reasons for this, but at least based on feel, it seems that one of the reasons is steadily dropping effectiness of small ship tactics. For instance, a common tactic that light ships can use is the hit-and-run attack aimed at whittling down opponents. For example, racing in to pound away a target's shields, do a little hull damage, and then run away to heal the damage I've taken. Then repeat the process. So long as I have faster regen than my target while having comparable damage, I can eventually cut them down even if my total amount of health is significantly lower than theirs. 

This works great in the early game, especially when hull repair is basically non-existant. Once the enemies that the player encounters start having hull repair and/or both higher shield regen and shield capacity; then hit-and-run tactics become useless.

The main factor in this, I think, is the heavy reliance on regen being the only way to change a ship's durability. 

Now, I think it is perfectly reasonable that some ships are equipped to win the the "death by a thousand cuts" style of combat. But in our current paradigm, it seems that virtually *every* ship is built for this due to a lack of defensive alternatives. In fact, there is a real lack of variety in defensive/survival oriented outfits, and in fact are pretty much limited to:
- Anti-missile turrets (takes up a turret slot)
- Missile Jammers (largely considered to be ineffective)
- Shield regen (exists in large variety across virtually every faction)
- Hull regen (universally present in certain areas, but fairly rare otherwise)

In counterpoint to this the Escape Velocity series added several additional outfits inconsistently across the series:
- The Flare Dispenser (small device that dropped flares that each individually neutralized a single heat-seeking missile. Very powerful against that specific munition, but useless against anything else)
- Shield Capacitors (small outfits (generally in the 5-10 range) that add a small amount of shields)
- Armor Plating (small, or sometimes variable sized outfits that added a small amount of armor. Some games these were just like shield capacitors (ex. something along the lines of "take up 3 space, get 200 hull points", others were variable, ex. "takes up 1% of the hull mass of the ship in outfit space, adds that much mass, and adds 300 hull points.")

Note that as best I can recall, EV classic had: Jammers, flare dispensers, shield regenerators, shield capacitors, and hull plating.
EVO lost the flare dispensers, but gained AM. (Might have also gained hull regen, but I don't recall)
EVN had everything except flares., with what felt like a decent number of options (at least 1 shield capacitor, sheild regenerator, armor plating, and AM available in most factions)

One thing that is worth noting is that in the EV series numerical limits on outfits was a thing. As in, most of these outfits had a hard capacity limit that the player could not install more than 2 each. Of course, the EV series was almost as easy to mod as ES is, and plug-ins that increased these limits were commonplace. I personally just edited the limits out of the datafiles; but even given that rarely exceeded where the limit would have been anyway.

A key difference (at least, from what I remember) is that the number of regen outfits that were capable of making a significant difference *during* sustained combat were slim. I think the Polaris ones in EVN were good enough to provide a noticeable difference during a capital ship slugging match; but by and large across the series, the difference between "good" regen and "bad" regen was basically just the difference in how long one had to stay out of direct combat while one waited to recover. Bad regenerators, one had to hang out in deep space for many minutes to recover one's shields. Good regenerators one merely had to be out of combat for several minutes to fill up one's shields again. In such an environment, a small ship with a decent regenerator is quite capable of dodging in, laying on some damage, then running away to recover, then repeating the process; and eventually wearing down the target.

Compare that to ES, where even some of the early human regenerators brag of being able to outpace weaponry, and letting up the damage in order to retreat and heal usually results in the target being fully healed too. This means that agility (aka being able to control the engagement and disengage at will to heal is devalued, as disengaging is just as likely to benefit one's opponent as oneself. As such, staying power in order to maintain the engagement and continue dealing damage (aka. Slugging it out) is heavily favored as the main combat style of significance, with kiting being strongly favored as well if one can get a range advantage. 

Within the ES paraidgm, since slugging it out is heavily favored as the main form of combat, it follows that the best ship to fly is the one with the best capacity for regenerating under fire; preferably with the largest amount of shields to cushion sudden impacts. This is virtually always the heaviest warship available, and against some targets basically requires multiple warships to do with any sort of speed, dependability, or confidence.

As I recall it, throughout the EV series I routinely faced a tactical choice in outfitting my ships. Assuming I had space to install any defensive outfits, I faced a choice due to the effectiveness of the outfits. If I was confident in being able to dodge the munitions, then shield regenerators would generally serve me best as it helped me recover faster between bouts and thus was better able to conduct hit-and-runs. On the other hand, if I was dealing with a target who was particularly agile (or I was particularly slow) and unable to count on dodging munitions, then opting for either shield capacitors or hull plating would yield the better investment, as these would enable me to survive large single-point damage spikes. (All the regen in the world won't help even a smidgen if one gets hit by a weapon with sufficient damage to destroy all of my shields AND all my hull, and heavy rockets - particularly in EV classic - were terrifyingly effective at dealing large amounts of damage). Beyond that, shields tended to be lighter, but more weapons were better against shields, so armor had its own strengths against them.

Beyond that, Jammers were actually quite effective, particularly against swarms of missiles (I don't recall any of the AM being particularly *good* against swarms, although more advanced ones generally did a bit better).

So there was both a healthy competition for my space, and a variety of tactical implications that contributed. The end result being:
- Light ships were valid and useful solo ships much longer into the game.
- More useful and effective options to use my outfit space.
- More diverse strategies, each of which tended to benefit more from different defensive outfits.
- More diverse roles for ships. Since not all ships had large amounts of regen, some would be better for slugging, others for raiding, others for kiting, etc. This resulted in larger variations between ships, significantly fewer death balls, and more strategic combat.

As a result of all this, I would like to suggest that we take a good long look at our regen values, and strongly consider how shield capacitors and hull plating could be incorporated into the game in a reasonable way.

**Comment by brendanjones** [comment]
I can't agree with this enough. It makes me sad that light and medium ships are useless beyond the early game in ES, and play styles are limited. Remember being able to play EVN late-game in a rail-gun Starbridge? Where's that kind of versatility in ES?

To add to everything @Zitchas said:
I wouldn't only look at nerfing regen values. For high regen outfits you could up the energy requirement, requiring higher overall energy production and/or storage on your ship. Given the energy production and storage outfits generally have high mass, having to outfit more of them in effect limits the amount of charging with which you can outfit yourself. Or you could choose for high charging, but at the expense of other outfits like, say, weaponry. (Though any decent sized warship can fit at least one Quarg Skylance. The mass of those should be increased and the damage decreased, they're massively overpowered as is. It's literally the only weapon worth having in the endgame, and that sucks.)

Another route to take is to have weaponry be more specific to shield or armour damage, and alongside that have more ships with higher armour than shields. For example, remember trying to use Crescent weaponry against Voinian ships in EVO? Yeah, those energy weapons were useless against the Voinian armour. 

So nerf energy weapons to do way less damage against armour, and give some races/factions a ship characteristic of having more armour than shields. The Korath are perfect for this, especially the Kor Sestor and Kor Mereti; they literally have hull regeneration tech already so their ships having high armour is logical. e.g. How about flipping the shield and armour values of a Korath Raider so that they have 9K shields and 27k armour? 

Another attribute to give more variety to play styles (which Zitchas touched on) is weaponry range, firing rate, and tracking speed (on turrets). The 'gun' type weapons (lasers etc) of every race are essentially the same, with the major difference being in their damage output. There are a couple of ones with differences, like the ion cannon and Kor Sestor weapons, and I love the differing attack styles these produce, but outside of Hai and Korath space the fighting styles are all practically the same. Give me a super powerful short-range weapon for ramming. Give me a longer-range weapon with slower firing rate. Give me some slower tracking turrets that I could actually avoid with skillful flying in lighter, faster ship. Just a few ideas, but they would already produce some different fighting/flying styles.

**Comment by brendanjones** [comment]
hahaha @Zitchas you're a hypocrite! You gave yourself a ship that cloaks and regenerates once your shields get too low. So much for "too much regeneration". Come back and fight me! I'm getting more ships with more Skylances, next time you're going to be toast, buddy.

<img width="1792" alt="Screen Shot 2022-07-22 at 20 08 01" src="https://user-images.githubusercontent.com/16049594/180502434-10df048e-d5d2-4a2a-97f0-39d30b093dd1.png">

**Comment by Zitchas** [comment]
Lol. It exists within the current paradigm, can't say anything more or less than that, really. The Heron isn't a warship, it is a ship built to *survive.* The entire point of it is to escape, run, and survive. Everything else is secondary. All those drones and fighters? They are just cover to provide the Heron with the opportunity to escape. Not to mention the fact that it has a scram drive (alongside regular hyperdrive and jump drive) to ensure that it is *very* hard to pin down...

That aside, why are you trying to kill it in the first place? It's background decor, not something that needs to be killed.

That being said, my person ship is just a placeholder. Once the Heron is used in the story (most likely with vastly changed stats), I'll likely be changing my person ship to something less obtrusive and spoilery. A Starbridge, maybe. Or perhaps a Penguin. I love those two ships. A Peregrine is also a strong possibility. In any case, almost certainly a much, much smaller ship than the Heron.

In any case, it perhaps simply reinforces the fact that regeneration is king in Endless Sky.

**Reply by brendanjones** [reply]
> That aside, why are you trying to kill it in the first place?

It attacked me! Not sure why it was aggro; maybe I attacked a previous version of your ship or something, because I'd never seen this one before. That said, I would probably attack it anyway, just to see what happens. You never know what kind of stats these author ships have and it's fun to see what happens ... and to collect them when they're disable-able ;) 

My fav discovery so far is the shooting star
<img width="763" alt="Screen Shot 2022-07-23 at 19 08 27" src="https://user-images.githubusercontent.com/16049594/180615485-010ae4f8-699f-4bcc-b27a-428a66ecbee7.png">

**Reply by Zitchas** [reply]
Ah, that might do it. "Author" is a government, and like most governments one can develop animosity with them.

The Shooting Star is awesome, agreed. Have you tried killing it?

**Reply by brendanjones** [reply]
Aha, I didn't know Author is a government, TIL. Yes I captured  the shooting star. My escorts had disabled it before I knew what was happening so I figured I might as well. Do those unique outfits it has have any effect if you equip them on your own ship? And is there any way to pilot it as your own ship?

**Reply by Zitchas** [reply]
They are functional outfits, yes. I don't recall off the top of my head if Beccabunny did anything to them to make them un-removable. They *may* have the "you can use the outfit, but only if you sell the ship" setup (like the cloaking device on the Arfecta that you can't actually remove, but can get if you scrap the ship.)

**Reply by lumbar527** [reply]
> The Shooting Star is awesome, agreed. Have you tried killing it?

Yes I have, with the best interceptor (Merganser). The kitting-power on that is OP!

**Comment by warp-core** [comment]
>So long as I have faster regen than my target while having comparable damage

So, you want to fly a smaller ship (less durability and less space for weapons, outfits, and engines, but less mass and drag so faster?) with comparable damage (same weapons? not sure how you'd fit those on a smaller ship, better weapons, then) and faster regen (so, again, better regen outfits, higher tier?)?

**Reply by brendanjones** [reply]
I think Zitchas just means if you preference shield regen tech then you can guarantee winning against a ship with similar damage output, because shield regen is so high in ES. That’s not an opinion or a desire, it’s how it already is.

**Comment by Pilcrow182** [comment]
Pardon if I'm just not understanding, but I don't think simply adjusting the existing regen / power draw values would really fix the problem here. Even if bigger shields are made to have less regen or take more energy than they do now, the smaller ships will still have less-capable shields than the bigger ships, and less outfit space to _power_ the bigger shields.

I think the better idea would be to add a regen delay timer that gets _reset_ every time a ship is hit (maybe as a fraction of the shield's regen speed itself?). Smaller shields would have less (maybe even zero) delay, and thus _start_ regenerating sooner after being hit, at the expense of having slower regen speeds. Larger shields would wait for longer before starting their regen, but then regen much faster once they do start. This way, a smaller ship can run in, get hit while dealing damage, run out to recover, and then run back in to attack a bigger ship before its shield delay has resolved (interrupting the larger ship's regen in the process).

This would add a tactical element to large ship customization too: you'd have a choice between having _big_ shield generators with faster regen but more delay, or stacking _smaller_ shield generators to try to get the same recovery speed with less delay (at the expense of outfit space). Or maybe stacking two different sized shields for a _staged_ regen setup: slow recovery shortly after being hit, with faster regen kicking in after more time has passed.

**Reply by Zitchas** [reply]
Yeah, the whole situation is not a quick fix. Your idea is definitely a good candidate for diversifying things.

**Comment by MasterOfGrey** [comment]
One problem that I think we have with combat balance, which I allude to in #6481, is that there's just not enough scale on combat effectiveness to work within.

The combat rebalance includes a line in the philosophy point: `"The combat matchup philosophy for how much outfit space is required for a ship and such works on the basis that a Heavy Warship should be able to support enough shield regen so as to completely, or nearly completely, ignore a fighter totally equipped with guns and guns only."`

Regen is king, but it's king because we don't have the supporting systems to make it good when it should be and bad when it's not.

We can't make super-deadly torpedoes that don't actually threaten small ships, cause big ships move way too fast so the torpedoes have to be able to track fast ships - making them deadly to small ships.
We can't create weapons that are meaningfully specialised against small ships, because big ships become too vulnerable using them.
We can't incentivise countermeasures and other forms of non-pew-pew vs pure regeneration combat, because small decreases in regeneration have disproportionately large impacts on balance in a fight - losing regeneration to negate or greatly reduce one damage type isn't worth it because of the proportional loss vs all the others.
And small decreases have a large proportional impact because big ships don't have the space to really carry enough regeneration to set them apart.

Essentially we have a game of small ships, and small ships that can put on enough regeneration to make them the best small ship.
We lack systematic, mechanical, diversity.

**Reply by brendanjones** [reply]
> We can't make super-deadly torpedoes that don't actually threaten small ships, cause big ships move way too fast so the torpedoes have to be able to track fast ships - making them deadly to small ships.

Yup. Ship speed and manoeuvrability need better balancing: https://github.com/endless-sky/endless-sky/issues/5271#issuecomment-1491583216 and https://github.com/endless-sky/endless-sky/issues/5271#issuecomment-1491618286. 

This would help fighters avoid some forms of damage, which would sort out some of the regen issues. Currently fighters have to tank all damage just like heavy ships, which just isn’t right.

**Reply by Quantumshark** [reply]
I've proposed some changes in that direction in #8842, and suggested a framework for a larger-scale reworking of ship mobility in #8957. If both of those were implemented, fighters would have a major mobility advantage over other ships.

**Reply by Saturnalia9** [reply]
> I've proposed some changes in that direction in #8842, and suggested a framework for a larger-scale reworking of ship mobility in #8957. If both of those were implemented, fighters would have a major mobility advantage over other ships.

Not to mention, we'd not have that "heavy warship is just better in every way" problem we've had since the dawn of time.

**Comment by leklachu** [comment]
It feels to me like part of the problem is the balance and lore/story are designed for the low end of what's available. Big ships are indeed slow because they're heavy... until you discover and afford atomic engines. They're also too pricey, so you fly with what you can get... until you grind up higher. The idea of a little fighter being able to take out a dreadnought is what you might want as a player, because you're special, but not what the 'world' expects. If you could only scrape together a few million in your lifetime, you couldn't dare challenge a huge warship (unless you're really special, like some players here far more skilled than I), but perhaps you've got more variety to find your place to fit in the galaxy.

Same goes for cargo runs and salaries: they just don't mean as much if you're capable of reverse-piracy. And all this goes even more-so when you get alien tech.

I dunno, I kind of like the focus being you're not the superhero who destroys everything in the galaxy at a touch... but then I lose escorts in a big fight 'cause they won't fly away when they're hurt, and I end up in this loop either of "fighting is for other captains" (I even did a pacifist run up to a point!) or "if I just had a few more heavy warships none of them would get killed": and my run becomes farm multiple Kar Il Vok 349s before starting the main storyline. ("Hi, I don't really need your salary, because all my fleet are run by robots and my flagship has an in-built cloak.")

So I treat balance, like exploration, as mostly part of my first run: and I stick around for the story and relaxation. There's only so much you can 'fix' a game without turning it into a completely different game.

I don't know if that helps this discussion.... but maybe if there's ways to help players find their niche in the galaxy without being the greatest possible money or piracy can buy. And making that feel rewarding or necessary to some. E.g. @brendanjones on skylances as "the only weapon worth having in the endgame" - but because my character usually refuses to fight Quarg without provocation, I'm constrained to lesser weapons (disruptors and slicers are fine, though).

**Reply by brendanjones** [reply]
Altering ship speed and manoeuvrability is definitely part of solution to some of the issues described in this thread regarding lack of variety between ships.

For example: https://github.com/endless-sky/endless-sky/issues/5271#issuecomment-1491583216

**Comment by ChuckStarchaser** [comment]
Quote: "The idea of a little fighter being able to take out a dreadnought is what you might want as a player, because you're special, but not what the 'world' expects."  That's true up to a point.  In the real world, a small fighter can indeed take out a cruiser, with the right weapon, as illustrated by the sinking of the Moskova, recently.  Well, not exactly;  it was a number of "small" assets working together.
But in any case, if we are discussing the lack of strategic variety in the game, the fact that bigger ships trump smaller ones is merely illustrative of the problem.  I feel the same, and I don't have much to offer in the way of a solution.  What I can do, however, is point out to games that have in some way NOT had the same problem.  One of my favorite games of all time was Masters of Orion II.  In it you actually designed your ships, and there was a staggering number of weapons and systems.

  Early in the MOO2 game it made sense to design and build just missile frigates.  However, by the time the Antarans would start their random attacks, it became necessary to design a medium sized ship specialized on disabling and boarding enemy ships, as Antaran technologies were awesome and you could not get them any other way.  Later it became necessary to design a tiny frigate of a ship without weapons but with systems that benefitted the whole fleet, such as the scout-lab, which improved your fleet's abilities against monsters, the warp dissipator, which prevented enemies from retreating and escaping, and eventually the fleet-wide missile jammer.  Eventually you added large ships with beam power galore, but smaller ships had their niche uses alongside.

  The core thing was the need to design specialized ships:  Outfits take up space, and if you try to design a ship that has both missiles and beams, for example, you could not fit all of the outfits that benefitted missiles, such as the fast missile rack, AND all of the outfits that benefitted beams, such as better aiming computers, high energy focus and structural analyzer.  It was better to design a good missile ship here, and a great beam ship there, than to try to mix things.  This idea could fairly easily be implemented into ES.  Just come up with new outfits galore, a dozen of them to help the effectiveness of rockets and missiles;  and a dozen of them to improve beam accuracy and quality.  And make sure that there are 5 times as many outfits in total as you could fit in any ship, even the largest.  This would force specialization, which is a good thing in itself;  but it doesn't address the USE of such specializations.

  In MOO2, the combat was turn-based, so you had the time to think and tell each ship what to do.  So, you made sure that each ship did what it was best at doing.  In ES we don't have the luxury, as combat is so fast-paced I, at least, have to rely on my fleet to do the fighting for me.  And I think this is the bigger problem.  I generally hate fast-paced anything, precisely because it ends up becoming a dumb, single strategy, bigger beats smaller paradigm.  But I like the story, so I keep playing.

**Reply by brendanjones** [reply]
This is of course dependent on fitting within the current vision for the game, but I think you’re onto something with outfit upgrades. As in, outfits whose purpose is to add to or change the attributes of another outfit. 

Some already exist in game actually: I forget which race it is (maybe Gegno?) and the exact attributes they change but iirc there are nanite(?) modules that increase energy generation of another unit you have installed(or was it shield generation?). Anyway, the point is the mechanic already exists. 

We could have range extenders of guns. Tracking rate enhancements for turrets. Shielding for missiles. Fuel efficiency boosts for hyperspace jumps. Active shield generation that, on command (not passively) burns loads of energy to divert power to shields. Expansions to the number of gun or turret slots (these existed in previous EV games). Hull reinforcement to add to armour. 

You get the idea, those are just a few ideas. The point is that you can use these upgrades to specialise a ship. Imagine a medium warship outfitted to focus purely on range-extended guns so that it can engage capital class ships by kiting them. For example.

**Reply by Saturnalia9** [reply]
> Expansions to the number of gun or turret slots (these existed in previous EV games). Hull reinforcement to add to armour. 

The devs have already vetoed this because there's no way (currently) to assign a new turret/gun hardpoints. For instance, the 256 and 512's default hardpoint for a new gun or turret would be 0,0. You'd be placing a weapon in empty space.

**Reply by brendanjones** [reply]
Good to know. Is it technically possible to alter the attributes of other outfits, like some of the suggestions I made? For example, could you extend the range of a given laser gun (e.g. a Heavy Laser) by 100?

**Reply by leklachu** [reply]
Weapon range is hard-coded in the data files as projectile velocity and projectile lifetime. I don't think you could have an outfit that modifies the range of another outfit, without changing the program code. Moreover with beam weapons, the 'projectile' sprite is fixed at the right length of pixels - even if you modify the range by hand, you have to lengthen the image to take the whole distance from ship to max-range, or it leaves a visual gap.

**Reply by leklachu** [reply]
Hull reinforcement is possible: there are stats for damage reduction of various types (but it could be easy to unbalance! The guide already advises against outfits that add shield, not shield regen.)

The others, I can't see a way of doing except for having a different outfit/weapon with the better stats. The way the Bunrodea  moddable power plants and shields work is they have space within them for the mods; then the mods themselves add energy or heat or regen or whatever. The 'boosters'/'limters' don't actually modify the base, they're plain outfits themselves.

**Reply by brendanjones** [reply]
Gah, as I feared; what seem like simple implementations have lots of technical blockers!

**Reply by Saturnalia9** [reply]
> Good to know. Is it technically possible to alter the attributes of other outfits, like some of the suggestions I made? For example, could you extend the range of a given laser gun (e.g. a Heavy Laser) by 100?

Lasers cannot be edited, simply because of the sprite. The projectiles, on the other hand, could possibly be changed. It would take some crafty text files, but theoretically you could redefine a weapon's velocity or lifetime using ammunition.

@AvianGeneticist made a weapon that required a battery to operate. That battery not only gave the weapon infinite ammo (as opposed to the default finite amount), but there were other outfits that modified the weapon further.

**Reply by pepoluan** [reply]
w.r.t. the original thoughts about "specialized ships"...

I'm currently doing a run wherein I actually ended up doing that. But rather accidentally... See I managed to cap a Marauder Levy and a Marauder Falcon. The latter as usual get specced to be a hotrod, while the former... well, no matter what I did it stays a lumbering giant.

So I specced the Falcon with short-range 'tackling' weapon, while I just upped the Levy's turning ability and specced it with 'sniping' weapons.

This strategy actually works quite great!

The challenge is of course to be able to cap the right kind of Marauder ships...

**Comment by yafflehk** [comment]
Unless you make it lore that guns and turrets have to be outside of shields and can be targeted and damaged separately from the main ship. Then a swarm of single fighters (or the players heroic Starbridge) can really influence a battle by taking out weapons, leaving the main ship vulnerable.

**Comment by leklachu** [comment]
That's an interesting idea. Might make for a way for player to take damage that's not as infuriating as losing one of your fleet that sat in battle too long, too. But it would risk overcomplicating gameplay. Overall I don't think the gameplay clutter would be worth it.

I had an idea for turrets, given how they sit out, visually, on a turret mount: that they could take much less outfit space, but increase drag.

I suppose, your swarm of single fighters - isn't that what the Hai do with small craft with ion cannons? They scramble the target's weapons, as well as draining its power. Maybe there could be a post-hai-reveal development where new ion weapons get developed in human space, and become a dominant tactic.

**Reply by Saturnalia9** [reply]
I tried that in my artillery PR. The devs won't allow drag to be visible on an outfit. And since the stat can't be hidden, it's a no-go.

**Comment by Zitchas** [comment]
Even without using ion or special damage types, merely giving fighters strong amounts of lateral thrust and AI that gets them to circle the target ship and focus on attacks on the rear (and thus avoid the front where the guns are) does wonders for both helping them survive longer and feel more effective in harassing big ships. They can definitely pin down big ships doing that.

**Reply by Saturnalia9** [reply]
To get back to what you said at the very beginning, what about weapons that slowed down shield regen and/or hull regen? Disruption essentially nullifies shield as a whole, and current weapons indirectly lower shield regen just by dealing pure damage. But if fighters got weapons specific to fighters that slowed shield regen by a percentage per second, that could address the initial problem.

As for interceptors... No idea.

And what you said about thrusters that help the ship avoid the front, we have that persistant problem where turrets are just universally better than guns. Their turn rate is incredibly high across the board, their range is the exact same as their gun counterparts, and they only use up slightly more energy than their gun counterparts.

Making turrets more specialised would help smaller ships survive. Suggestions:

- Big damage, snall range
- Big range, high energy
- Big burst, big range, high energy

Something along those lines.

**Reply by Zitchas** [reply]
I suppose the problem is that right now, all our turrets are just "guns mounted on a swivel platform". Usually multiples, at that.

**Reply by mOctave** [reply]
Yes, it seems that turrets fall into one of two categories:
1. More guns than you can mount on your interceptor/light warship, all in one place, that can turn.
2. Highly effective equipment that destroys your useful missile weaponry, that you can't afford to install to block the enemy's.

**Reply by Saturnalia9** [reply]
> I suppose the problem is that right now, all our turrets are just "guns mounted on a swivel platform". Usually multiples, at that.

When something several tonnes can instantly swivel 160° and lock onto the centre of mass of another ship, all in one second... Unless the energy consumption is incredibly high, there is no weakness to that.

**Reply by Quantumshark** [reply]
Turrets generally tend to have higher damage per hardpoint at the cost of lower damage/ton (which makes sense for a weapon designed for use on large ships, which have more weapon space per hardpoint) on account of frequently being "2-4 of X gun, plus a mounting".

> I suppose the problem is that right now, all our turrets are just "guns mounted on a swivel platform". Usually multiples, at that.

Is "guns mounted on a swivel platform" not just the definition of a turret? Unless you mean them being turreted versions of weapons that also exist as guns, but that's certainly not the case for every turret - off the top of my head, there's no gun equivalent to the Disruptor, Skylance or Grab-strike, and certainly not of AM turrets.

More generally, I think it makes sense for many turrets to be designed as a larger ship's defence against smaller ships, simply because that's the most logical role for them.
You can certainly have big artillery turrets, but if you're deliberately specialising a weapon against large targets that can't dodge effectively a rotating base is just unnecessary mass, and you'd do better with a big fixed gun like a Thorax Cannon or Ka'het Nullifier.

I'd also be cautious about lowering turret traverse rates, because lower turret traverse weakens them in the player's hands much more than it seems to when they're used by the AI, and we should avoid a situation where we have a weapon that's strong in the AI's hands but useless for players, because that's profoundly unenjoyable.

I recognise that this is likely to be a controversial suggestion, but how about making the AI _worse_ at using turrets? I'm not really sure how best to go about it, but it does seem like the logical way to resolve my previous point.

**Reply by Saturnalia9** [reply]
Only way I can think of that would worsen AI use of turrets is if the AI was recoded to have the personality to spread out to encircle the target(s). Rather than all enemies beeline straight for the target.

That way any AI vs AI would be forced to focus one at a time, which would inherently make them take more damage during the process of readjusting. Meanwhile, the player can use opportunistic (advantageous against an encircling strategy... sort of).

**Reply by mOctave** [reply]
Alternatively, we could use the confusion number attached to an NPC's personality and use it to reduce other attributes as well (ie turret traverse = base turret traverse / (1 - (confusion/200)) ).

**Comment by leklachu** [comment]
Next up: Train KataGo to fight ES fleet battles.

More realistically... Do we have plans in the works for improved AI? Perhaps it's too much to hope for, but I'd love to set up a fleet with defensive tactics to hit-and-run into the edge of battle.

Come to think of it, that's kind of what the Remnant appear to do with cloaking, cloak out of battle when their health runs low, and rejoin when ready. Is that with any special settings? Perhaps I should take a look at that! Maybe you've just inspired my next fleet...

**Reply by Zitchas** [reply]
Some people have worked on better AI tactics in the past, but no broad "AI" in general.

**Comment by Quantumshark** [comment]
For fighters specifically, their strength should come from numbers, they shouldn't be particularly powerful individually. Maybe a large ship does have enough regen to completely ignore damage from one fighter, but it certainly doesn't have enough to do the same with ten. The whole point of a fighter is it's something that's deployed in groups. One fighter should not be a significant threat to anything, just like one ant isn't a threat to anything. But a full wing of fighters deployed from a carrier? _That_ should be scary, just like a full colony of ants is scary. Part of the problem we have is that our fighters are too big relative to their carriers, and thus carriers can't carry very many of them because it completely flattens their handling, and part of the problem is that our carriers are just warships with a few extra fighter bays slapped on. 

The ideal balance point would be one wherein one carrier with a full wing of fighters represents a side-grade from one non-carrier warship, with both having relative advantages and disadvantages and offering a different gameplay style, and neither being objectively the better option.
In our current state, a carrier without its fighters is generally already pretty competitive with a non-carrier warship, but adding the fighters doesn't change much. This is, technically, balanced, but it doesn't lend itself to variety in gameplay style - a 'carrier' is currently just a warship with a "lose 2 million credits" button. The Navy Carrier, to use the most egregious example, is already the strongest human warship without its fighters, which gives the fighters no room (negative room, in fact) to be good on their own.
As an alternative design proposal for carriers, how about this: give them more engine space, much less weapon space, and much heavier hulls? Making carriers heavier while giving them more engine power to compensate would let them carry a lot more fighters without causing handling issues, and reducing their direct combat power would create much more room for their fighters to be useful without breaking the overall balance of the game.

**Reply by Saturnalia9** [reply]
A fighter generally can hold 1 or 2 Energy Blasters. You'd need... what, 3 energy plasters to begin to compare to a Proton/Particle. Provided a heavy warship can fit between 6 and 10 of these, you'd need a carrier that can carry between 18 and 30 fighters to be able to compete equally (if not slightly lower) than a current fully-outfitted heavy warship.

On the other hand, if you have a fleet consisting of one heavy warship and 2 medium warships, you'd need between 80 and 100 fighters to take that fleet out, simply because you'd be losing fighters (and DPS) considerably faster than the enemy is losing DPS. 

So maybe have a single carrier be able to hold between 22 and 40 fighters.

**Reply by Quantumshark** [reply]
Most human fighters can actually carry three energy blasters, and that's far from the ideal build for them. The best setup currently, as far as I'm aware, seems to be to fit the fighters out with missile pods and have them rearm from the carrier. That gives them much better DPS than you'll get with energy blasters, and the carrier's ammunition reserves allow them to sustain that DPS. With the right build, even just ten fighters can be pretty threatening.

**Reply by lumbar527** [reply]
I don't use missiles for the same reason some people don't use fighters: they cost money to replace. So if you have to use missile loadout with them, then fighters have to be an investment no matter what.

(Also I like to loot enemies)

**Reply by Saturnalia9** [reply]
@lumbar527 Same reason I don't use missiles. I don't like having to pay just to fight.

**Comment by VaelophisNyx** [comment]
there is always the solution you could look into, of adding specific damage resistance outfits or hull-stats, such that there's more tactical nuance than just eating 5k damage and regenerating it on the fly.

personally I feel hull regen should be restricted to very high tier alien races, and organic ships. Otherwise damage is not scary until its overwhelming.

**Reply by Saturnalia9** [reply]
So you mean nerfing shield/hull regen and adding resistances? (Just so we're on the same page.)

Hull regen is already restricted to fairly high-tier alien ships, unless you have a different way of counting tiers. 🤔

**Reply by VaelophisNyx** [reply]
Just hull regen being restricted, not nerfed. Technically speaking, you can get it as early as T1-1.5 ships over with the Coalition, requiring only a jump drive. Otherwise, it's not all that hard to get in general and is easily slotted into just about every ship in the game.

Shield regen is fine, since they're made to do that and hard-counter weapons exist to disrupt or pierce shields.
I think only specific race's ships should be hull-regeneration capable, and that others should be able to add some form of specified damage resistance as an outfit in its place (ie: hull damage resistance for the trade-off of being bulky, heavy, and reducing cooling efficiency).

**Reply by Quantumshark** [reply]
The problem with that is that once the player gets access to any (non-ammo-limited) source of piercing or disruption, any opponent without hull repair becomes trivial to beat by simply inflicting chip damage to their hull and then retreating to heal, slowly wearing them down.

**Reply by Zitchas** [reply]
That sounds like good tactics. It's good to have some ships built to counter this tactic. It's bad when *all* ships are built to counter this, as it contributes to the current meta where the primary two viable tactics are "have biggest ship and slug it out" followed by "if you are lucky maybe you can kite them". Everything being a slugfest devalues skilled flying, devalues mobility, devalues thought, strategy, and tactics. It even devalues slugfests (which can be fun on occasion).

Having combat-relevant regen should be an explicit choice to complement a particular strategy, not a given that all ships will have lots of it as is currently the case. More regen should equal less other stuff (like weapons). Even more so with hull regen.

**Comment by Hurleveur** [comment]
I'd like to point out that some weapons such as the tripulse shredder or melee torpedoes allow ships to take advantage of their mobility advantage and effectively gain dps (because dps only matters when you are fighting)
Small ships thus stay viable and can gain better dps by using more appropriate weapons as the game goes on. We could use more weapon diversity, in more ranged options - and radar jammers should ofc be better (or just have alien versions)

Also an alternative to no hull regen would be to simply reduce the max hp as hull is repaired. The uhai hull repair will have a delay applied on it
